pkg/rgrpc/todo/pool/selector.go

```diff
@@ -130,9 +132,15 @@ func (s *Selector[T]) Next(opts ...Option) (T, error) {
 
 	existingClient, ok := s.clientMap.Load(target)
 	if ok {
+		if options.logger != nil {
```

Comment:
We should not allow `nil` loggers. The `logger` field should default to a noop logger.